### ISSUE TITLE
Rewrite or remove /tutorials/server/httpserver

### ISSUE THREAD

**Issue by erf** [dart-lang/site-www#3058]
Page URL: https://dart.dev/tutorials/server/httpserver
Page source: https://github.com/dart-lang/site-www/tree/master/src/_tutorials/server/httpserver.md

Found a typo? You can fix it yourself by going to the page source and clicking the pencil icon. Or finish creating this issue.

Description of issue:

The `http_server` package is used in one example, but it looks like its pretty much abandon ware by now?

**Comment by kwalrath** [comment]
@kevmoo or @sigmundch what do you think we should do?

**Comment by erf** [comment]
Isn't the `shelf` package more relevant for "some higher-level buildingblocks"?

**Comment by kevmoo** [comment]
Yeah it would be a lot better if we talked about shelf here

**Comment by kwalrath** [comment]
It sounds like we have two choices, for now:

1. Update the example code and content to use shelf.
2. Remove this page and adjust other pages that refer to it.

I prefer the first option (in the short term, at least), because people do want this content. I took a look at 30-day and 1-year analytics, and this is consistently the #2 tutorial (after /tutorials/language/streams), and in the top-30 pages on the site.

In the short term, I'd love for someone to volunteer to update the page — or at least, to update the example code. (I know nothing about shelf!) In the longer term, we might want to consider creating a different kind of doc (set?) to cover this material. Tutorials have largely been supplanted by codelabs and (if we follow the lead of flutter.dev) cookbooks.

**Comment by erf** [comment]
Could you perhaps add some information on performance or best practices? 

A few maybe-not-so-releavnt questions: Should i use `Isolates` to handle each request? Do i call `flush` before `close` on a `HttpResponse` stream? How does `HttpResponse.persistentConnection` work and when should i use it? I also would like to see some more info on how to use `WebSocket.pingInterval`.

BTW: I really like Dart as a server language and i would like to see more community / ecosystem around it. Thanks!

**Comment by Sameerkash** [comment]
I was just taking a look at `shelf` as I was going through pub-dev, this will be a good opportunity for me, I'll try this one soon!

cc @kwalrath

**Comment by erf** [comment]
@Sameerkash Sounds good!

I just converted some pure Dart code to use `shelf`, `shelf_router`, `shelf_static` and `shelf_web_socket`. I was stumped a bit at first as to how use the latter two with `shelf_router`, but made it work in the end. Maybe a guide on how to combine these would also be helpful.

Maybe i could also contribute myself, but i'm very new to these libraries.

**Comment by Sameerkash** [comment]
> @Sameerkash Sounds good!
> 
> I just converted some pure Dart code to use `shelf`, `shelf_router`, `shelf_static` and `shelf_web_socket`. I was stumped a bit at first as to how use the latter two with `shelf_router`, but made it work in the end. Maybe a guide on how to combine these would also be helpful.
> 
> Maybe i could also contribute myself, but i'm very new to these libraries.

Anyway you can link what you wrote ?

**Comment by erf** [comment]
It's not open source, but i can paste the gist of it:

```
  final indexHandler = createFileHandler('web/index.html', url: '');
  final wsHandler = webSocketHandler(app.handle);
  final staticHandler = createStaticHandler('web');

  final router = Router();
  router.get('/', indexHandler);
  router.get('/ws', wsHandler);
  router.get('/<web|.*>', staticHandler);

  final handler = const shelf.Pipeline()
      .addMiddleware(shelf.logRequests())
      .addHandler(router);

  final server = await io.serve(handler, address.address, port);
```

**Comment by erf** [comment]
.. but my web app is here -> https://fakkt.app 😎

BTW: just found a great youtube tutorial here -> https://www.youtube.com/watch?v=ZKNKNxaliZQ 🔥

**Comment by Sameerkash** [comment]
> .. but my web app is here -> https://fakkt.app 😎
> 
> BTW: just found a great youtube tutorial here -> https://www.youtube.com/watch?v=ZKNKNxaliZQ 🔥

@erf Thanks, I was checking out the same tutorials since morning, one doubt I have is are you managing any ` js` on the client-side?

**Comment by erf** [comment]
Yes. A client request a index.html which loads some javascript, and then i talk to the server by sending messages via a WebSocket.

**Comment by Sameerkash** [comment]
> Yes. A client request a index.html which loads some javascript, and then i talk to the server by sending messages via a WebSocket.
> 
> BTW: i made my own dom lib called [elite](https://github.com/erf/elite), for building html with javascript.

Sounds great thanks !

**Comment by kwalrath** [comment]
Page has been gutted & now links to more info, but I think we still need replace the content.

**Comment by antfitch** [comment]
Tentatively closing this since the issue has gone stale for quite some time.

**Comment by Gustl22** [comment]
@antfitch the issue is still referenced in the code:
https://github.com/dart-lang/samples/blob/3a54a046b53f1456015877c27e2e2b43139998c8/server/simple/public/index.html#L43
